worker/worker_package.go

```diff
@@ -0,0 +1,108 @@
+package worker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+
+	"github.com/go-redis/redis/v8"
+
+	"github.com/opensvc/oc3/cache"
+	"github.com/opensvc/oc3/mariadb"
+)
+
+func (t *Worker) handlePackage(nodeID string) error {
+	_, err := t.Redis.HDel(context.Background(), cache.KeyPackagesPending, nodeID).Result()
+	if err != nil {
+		slog.Error(fmt.Sprintf("can't HDEL %s: %s", cache.KeyPackagesPending, nodeID))
+		return err
+	}
+
+	cmd := t.Redis.HGet(context.Background(), cache.KeyPackagesHash, nodeID)
+	result, err := cmd.Bytes()
+	switch err {
+	case nil:
+	case redis.Nil:
+		return nil
+	default:
+		return err
+	}
+
+	var data map[string]any
+	if err := json.Unmarshal(result, &data); err != nil {
+		slog.Error(fmt.Sprintf("unmarshalled data: %#v\n", data))
+		return err
+	}
+
+	if _, ok := data[`packages`]; !ok {
```

Comment:
Use back-tick "`" only for multiline strings
so here change:

`
	if _, ok := data[`packages`]; !ok {
to
	if _, ok := data["packages"]; !ok {
`

same comment for the wall file